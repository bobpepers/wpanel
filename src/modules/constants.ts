export const LOGIN_FETCH = 'LOGIN_FETCH';
export const LOGIN_DATA = 'LOGIN_DATA';
export const LOGOUT_FETCH = 'LOGOUT_FETCH';
export const LOGOUT_DATA = 'LOGOUT_DATA';

export const GET_USERS_FETCH = 'GET_USERS_FETCH';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_DATA_BY_FILTER_FETCH = 'GET_DATA_BY_FILTER_FETCH';
export const GET_USERS_BY_LABELS_FETCH = 'GET_USERS_BY_LABELS_FETCH';

export const GET_USERS_PENDING_DOCUMENTS_FETCH = 'GET_USERS_PENDING_DOCUMENTS_FETCH';
export const GET_USERS_PENDING_DOCUMENTS_DATA = 'GET_USERS_PENDING_DOCUMENTS_DATA';
export const FILTER_USERS_VERIFIED_DOCUMENTS = 'FILTER_USERS_VERIFIED_DOCUMENTS';

export const GET_CURRENT_USER_FETCH = 'GET_CURRENT_USER_FETCH';
export const GET_CURRENT_USER_DATA = 'GET_CURRENT_USER_DATA';
export const GET_CURRENT_USER_ERROR = 'GET_CURRENT_USER_ERROR';
export const CURRENT_USER_RESET = 'CURRENT_USER_RESET';

export const GET_USER_DATA_FETCH = 'GET_USER_DATA_FETCH';
export const GET_USER_DATA_SUCCESS = 'GET_USER_DATA_SUCCESS';

export const DELETE_USER_LABEL_FETCH = 'DELETE_USER_LABEL_FETCH';
export const ADD_USER_LABEL_FETCH = 'ADD_USER_LABEL_FETCH';
export const EDIT_USER_LABEL_FETCH = 'EDIT_USER_LABEL_FETCH';

export const CHANGE_USER_STATE_FETCH = 'CHANGE_USER_STATE_FETCH';
export const CHANGE_USER_ROLE_FETCH = 'CHANGE_USER_ROLE_FETCH';
export const CHANGE_USER_OTP_FETCH = 'CHANGE_USER_OTP_FETCH';

export const SIGN_IN_REQUIRE_2FA = 'SIGN_IN_REQUIRE_2FA';

export const ALERT_DELETE = 'ALERT_DELETE';
export const ALERT_DATA = 'ALERT_DATA';
export const ALERT_PUSH = 'ALERT_PUSH';
export const ALERT_DELETE_BY_INDEX = 'ALERT_DELETE_BY_INDEX';

export const USER_ACTIVITY_FETCH = 'USER_ACTIVITY_FETCH';
export const USER_ACTIVITY_DATA = 'USER_ACTIVITY_DATA';

export const ADMIN_ACTIVITY_FETCH = 'ADMIN_ACTIVITY_FETCH';
export const ADMIN_ACTIVITY_DATA = 'USER_ACTIVITY_DATA';

export const METRICS_FETCH = 'METRICS_FETCH';
export const METRICS_DATA = 'METRICS_DATA';

export const WITHDRAWS_LIST_DATA = 'withdrawList/WITHDRAWS_LIST_DATA';
export const WITHDRAWS_LIST_FETCH = 'withdrawList/WITHDRAWS_LIST_FETCH';

export const WITHDRAWS_INFO_DATA = 'withdrawInfo/WITHDRAWS_INFO_DATA';
export const WITHDRAWS_INFO_FETCH = 'withdrawInfo/WITHDRAWS_INFO_FETCH';

export const WITHDRAWS_USER_HISTORY_DATA = 'withdrawUserHistory/WITHDRAWS_USER_HISTORY_DATA';
export const WITHDRAWS_USER_HISTORY_FETCH = 'withdrawUserHistory/WITHDRAWS_USER_HISTORY_FETCH';
