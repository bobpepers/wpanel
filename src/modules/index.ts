import {combineReducers} from 'redux';
import {all, call} from 'redux-saga/effects';
import {
    alertReducer,
    AlertState,
    rootHandleAlertSaga,
} from './alert';
import {
    authReducer,
    AuthState,
    rootAuthSaga,
} from './auth';
import {
    changeUserReducer,
    ChangeUserState,
    rootChangeUserSaga,
} from './changeUser';
import {
    labelReducer,
    LabelState,
    rootLabelSaga,
} from './label';
import {metricsReducer, MetricsState} from './metrics';
import {rootMetricsSaga} from './metrics/sagas';
import {
    adminActivityReducer,
    AdminActivityState,
    currentUserReducer,
    CurrentUserState,
    rootAdminActivitySaga,
    rootCurrentUsersSaga,
    rootGetUserDataSaga,
    rootUserActivitySaga,
    rootUsersSaga,
    userActivityReducer,
    UserActivityState,
    userDataReducer,
    UserDataState,
    usersReducer,
    UsersState,
} from './user';
import {
    rootGetWithdrawInfoSaga,
    withdrawsInfoReducer,
    WithdrawsInfoState,
} from './withdrawInfo';
import {
    rootGetWithdrawListSaga,
    withdrawsListReducer,
    WithdrawsListState,
} from './withdrawList';
import {
    rootGetWithdrawUserHistorySaga,
    withdrawsUserHistoryReducer,
    WithdrawsUserHistoryState,
} from './withdrawUserHistory';

export * from './alert';
export * from './auth';
export * from './changeUser';
export * from './label';
export * from './user';
export * from './withdrawList';
export * from './withdrawInfo';
export * from './withdrawUserHistory';

export interface AppState {
    adminActivity: AdminActivityState;
    alert: AlertState;
    auth: AuthState;
    changeUserState: ChangeUserState;
    userActivity: UserActivityState;
    userLabels: LabelState;
    usersData: {
        selectedUser: UserDataState;
        users: UsersState;
        currentUser: CurrentUserState,
    };
    metrics: MetricsState;
    withdrawList: WithdrawsListState;
    withdrawInfo: WithdrawsInfoState;
    withdrawUserHistory: WithdrawsUserHistoryState;
}

const usersDataReducer = combineReducers({
    selectedUser: userDataReducer,
    users: usersReducer,
    currentUser: currentUserReducer,
});

export const appReducer = combineReducers({
    adminActivity: adminActivityReducer,
    alert: alertReducer,
    auth: authReducer,
    changeUserState: changeUserReducer,
    userLabels: labelReducer,
    userActivity: userActivityReducer,
    usersData: usersDataReducer,
    metrics: metricsReducer,
    withdrawList: withdrawsListReducer,
    withdrawInfo: withdrawsInfoReducer,
    withdrawUserHistory: withdrawsUserHistoryReducer,
});

export function* rootSaga() {
    yield all([
        call(rootLabelSaga),
        call(rootAdminActivitySaga),
        call(rootAuthSaga),
        call(rootChangeUserSaga),
        call(rootGetUserDataSaga),
        call(rootUserActivitySaga),
        call(rootUsersSaga),
        call(rootHandleAlertSaga),
        call(rootMetricsSaga),
        call(rootGetWithdrawListSaga),
        call(rootGetWithdrawInfoSaga),
        call(rootGetWithdrawUserHistorySaga),
        call(rootCurrentUsersSaga),
    ]);
}
