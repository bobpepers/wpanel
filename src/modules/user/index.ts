export * from './adminActivity';
export * from './currentUser';
export * from './userActivity';
export * from './userData';
export * from './users';
