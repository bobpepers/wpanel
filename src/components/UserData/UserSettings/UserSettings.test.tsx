import {shallow} from 'enzyme';
import * as React from 'react';
import { CurrentUserInterface } from '../../../modules';
import {UserSettings, UserSettingsProps} from './index';

const defaults: UserSettingsProps = {
    user: {phones: [], profile: {country: 'US'}},
    handleChangeUserState: jest.fn(),
    handleChangeRole: jest.fn(),
    handleChangeUserOTP: jest.fn(),
    currentUser: {} as CurrentUserInterface,
};

describe('UserSummary component', () => {
    const setup = (props: Partial<UserSettingsProps> = {}) =>
        shallow(<UserSettings {...{...defaults, ...props}}/>);

    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toBeDefined();
    });
});
