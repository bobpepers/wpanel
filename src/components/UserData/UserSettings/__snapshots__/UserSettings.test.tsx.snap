// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSummary component should render 1`] = `
<UserSettingsComponent
  classes={
    Object {
      "formControl": "UserSettingsComponent-formControl-2",
      "gridRow": "UserSettingsComponent-gridRow-3",
      "gridRowCenter": "UserSettingsComponent-gridRowCenter-4",
      "paper": "UserSettingsComponent-paper-1",
      "subtitle": "UserSettingsComponent-subtitle-6",
      "title": "UserSettingsComponent-title-5",
    }
  }
  currentUser={Object {}}
  handleChangeRole={[Function]}
  handleChangeUserOTP={[Function]}
  handleChangeUserState={[Function]}
  user={
    Object {
      "phones": Array [],
      "profile": Object {
        "country": "US",
      },
    }
  }
/>
`;
